src/v/cluster/partition_balancer_backend.cc

```diff
@@ -64,24 +64,55 @@ partition_balancer_backend::partition_balancer_backend(
   , _timer([this] { tick(); }) {}
 
 void partition_balancer_backend::start() {
-    _timer.arm(_tick_interval());
+    _member_updates = _state.members().register_members_updated_notification(
+      [this](model::node_id n, model::membership_state state) {
+          on_members_update(n, state);
+      });
+    maybe_rearm_timer();
     vlog(clusterlog.info, "partition balancer started");
 }
 
+void partition_balancer_backend::maybe_rearm_timer() {
+    if (_gate.is_closed()) {
+        return;
+    }
+    auto schedule_at = clock_t::now() + _tick_interval();
+    if (_timer.armed()) {
+        schedule_at = _timer.get_timeout();
+        _timer.cancel();
+    }
```

Comment:
Do we need a mutex to make sure that no two task will run in parallel ? Even if the timer is not armed that doesn't indicates the background task isn't running.